components/list/list-item-drag-drop-mixin.js

```diff
@@ -365,17 +366,25 @@ export const ListItemDragDropMixin = superclass => class extends superclass {
 		return this._dropLocation === dropLocation.below ? html`<div class="d2l-list-item-drag-bottom-marker">${renderTemplate}</div>` : null;
 	}
 
+	_getListSize() {
+		const parent = this.parentNode;
+		return parent.getListSize();
+	}
+
 	_renderDragHandle(templateMethod) {
 		templateMethod = templateMethod || (dragHandle => dragHandle);
 		const classes = {
 			'd2l-focusing': this._focusingDragHandle,
 			'd2l-hovering': this._hovering
 		};
+
 		return this.draggable ? templateMethod(html`
 			<d2l-list-item-drag-handle
 				id="${this._itemDragId}"
 				class="${classMap(classes)}"
 				text="${ifDefined(this.dragHandleText)}"
+				current-position="${ifDefined(this.dropDestination)}"
+				list-size="${ifDefined(this._getListSize())}"
```

Comment:
I think based on @dbatiste and our discussions, we should move instead to get the position and list size when the drag handle is focused. This will avoid having to explicitly pass the information down the tree in this way.